### PR TITLE
Handle no content in element tags

### PR DESCRIPTION
**Pull request by richtier**
- [x] Changelog entry added.
 - [x] (if there are vulnerable requirements) Upgraded any vulnerable dependencies.

### CODE REVIEW

File: tests/test_templatetags.py [hunk 1]
```diff
@@ -48,6 +48,30 @@ def test_add_anchors():
...
+                '<h2><b></b></h2>'
+                '<h2><b>Some title</b></h2>'
+                '<h2></h2>'
+                '<br/>'
+    })
+    html = template.render(context)
+
+    assert html == (
+        '<br/>'
+        '<h2><b></b></h2>'
```
Comment (eadpearce) [inline tests/test_templatetags.py:68]: shouldn't we be removing the tag completely if it has no content? i think empty tags are an accessibility issue

Comment (sdonk) [reply tests/test_templatetags.py:68]: It has content, inside the b

Comment (eadpearce) [reply tests/test_templatetags.py:68]: It does??

Comment (eadpearce) [reply tests/test_templatetags.py:68]: The one below does. This one doesn't

Comment (sdonk) [reply tests/test_templatetags.py:68]: Not this test, I mean the real case scenario is <h2><b>Foo</b></h2>

Comment (richtier) [reply tests/test_templatetags.py:68]: I think we should aim for behaviour parity with the browser. If the browser is happy then our code should be. Less complexity that way.

according to [the spec](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements), a heading element can contain a `<b>` (so called _phrasing content_), so it's valid to have a <b>

as for it being empty - the browser accepts that so I think we can justify not removing the empty b

**Pull request merged**